src/repositories/component/component-crud.repository.ts

```
@@ -0,0 +1,252 @@
+import { Component, ComponentInput, ComponentStatus } from '../../types';
+import { BaseComponentRepository } from '../base/base-component.repository';
+import { RepositoryRepository } from '../repository.repository';
+
+/**
+ * Repository for basic Component CRUD operations
+ * Handles create, read, update, delete operations for components
+ */
+export class ComponentCrudRepository extends BaseComponentRepository {
+  constructor(kuzuClient: KuzuDBClient, repositoryRepo: RepositoryRepository) {
+    super(kuzuClient, repositoryRepo);
+  }
```

Biome (1.9.4)

[error] This constructor is unnecessary.

Unsafe fix: Remove the unnecessary constructor.


(lint/complexity/noUselessConstructor)